src/main/java/de/maxhenkel/audioplayer/command/AudioPlayerCommands.java

```diff
@@ -375,6 +378,49 @@ private static LiteralArgumentBuilder<CommandSourceStack> applyCommand(LiteralAr
                                 })));
     }
 
+    private static LiteralArgumentBuilder<CommandSourceStack> bulkApplyCommand(LiteralArgumentBuilder<CommandSourceStack> builder, Predicate<ItemStack> validator, String itemTypeName) {
+        return builder.requires((commandSource) -> commandSource.hasPermission(AudioPlayer.SERVER_CONFIG.applyToItemPermissionLevel.get()))
+                .then(Commands.argument("sound", UuidArgument.uuid())
+                        .executes((context) -> {
+                            ServerPlayer player = context.getSource().getPlayerOrException();
+                            UUID sound = UuidArgument.getUuid(context, "sound");
+                            ItemStack itemInHand = player.getItemInHand(InteractionHand.MAIN_HAND);
+                            if (validator.test(itemInHand)) {
+                                processShulker(context, itemInHand, sound, null);
+                            } else {
+                                context.getSource().sendFailure(Component.literal("You don't have a %s in your main hand".formatted(itemTypeName)));
+                            }
+                            return 1;
+                        })
+                        .then(Commands.argument("custom_name", StringArgumentType.string())
+                                .executes((context) -> {
+                                    ServerPlayer player = context.getSource().getPlayerOrException();
+                                    UUID sound = UuidArgument.getUuid(context, "sound");
+                                    ItemStack itemInHand = player.getItemInHand(InteractionHand.MAIN_HAND);
+                                    String customName = StringArgumentType.getString(context, "custom_name");
+                                    if (validator.test(itemInHand)) {
+                                        processShulker(context, itemInHand, sound, customName);
+                                    } else {
+                                        context.getSource().sendFailure(Component.literal("You don't have a %s in your main hand".formatted(itemTypeName)));
+                                    }
+                                    return 1;
+                                })));
+    }
+
+    private static void processShulker(CommandContext<CommandSourceStack> context, ItemStack itemInHand, UUID soundID, @Nullable String name) {
+        ListTag shulkerContents = itemInHand.getTagElement("BlockEntityTag").getList("Items", 10);
+        for (int i = 0; i < shulkerContents.size(); i++) {
+            CompoundTag currentItem = shulkerContents.getCompound(i);
+            ItemStack itemStack = ItemStack.of(currentItem);
+            if (itemStack.getItem() instanceof RecordItem) {
```

Comment:
I'm pretty sure that if we convert this to check for both music discs AND goat horns that we'd instantly gain a bulk operation for them as well. Would just add a similar `goathorn_bulk` command in.

Thoughts?

Reply:
Personally I would keep them separate or it could confuse players... i.e. having a random goat horn in the shulker and expecting it to be untouched by the "music disk" command. However this is your mod and I'm totally open to whichever way you'd like it.

Reply:
Cool I'll work on pushing that up!

Reply:
Alright, made the additions. Was a bit more complex than I first thought, because checking for a RecordItem or InstrumentItem behaves the way you first mentioned. To keep the actions focused on only music disc or only goat horns I had to add an extra `actionItem` parameter to the methods.

Tested it out quick and works great